docs/user_guide/read_api/da_vinci_client.md

```diff
@@ -0,0 +1,52 @@
+---
+layout: default
+title: Da Vinci Client
+parent: Read APIs
+grand_parent: User Guides
+permalink: /docs/user_guide/read_api/da_vinci_client
+---
+
+# Da Vinci Client
+This allows you to eagerly load some or all partitions of the dataset and perform queries against the resulting local 
+cache. Future updates to the data continue to be streamed in and applied to the local cache.
+
+## Record Transformer
+This feature enables applications to transform records as they're being consumed and stored in the Da Vinci Client. 
+It's capable of handling records that are compressed and/or chunked.
+
+### Usage
+To use the record transformer, you will need to implement the 
+[DaVinciRecordTransformer](http://venicedb.org/javadoc/com/linkedin/davinci/client/DaVinciRecordTransformer.html) 
+interface and pass the object into 
+[setRecordTransformer()](https://venicedb.org/javadoc/com/linkedin/davinci/client/DaVinciConfig.html#setRecordTransformer(com.linkedin.davinci.client.DaVinciRecordTransformer)). 
+
+When a message is being consumed, the 
+[DaVinciRecordTransformer](http://venicedb.org/javadoc/com/linkedin/davinci/client/DaVinciRecordTransformer.html) will 
+modify the value before it is written to storage.
+
+Here's an example implementation:
+```
+import com.linkedin.davinci.client.DaVinciRecordTransformer;
+import com.linkedin.davinci.client.TransformedRecord;
+import com.linkedin.venice.utils.lazy.Lazy;
+import org.apache.avro.Schema;
+
+
+public class StringRecordTransformer
+    implements DaVinciRecordTransformer<Integer, String, TransformedRecord<Integer, String>> {
+  public Schema getKeyOutputSchema() {
+    return Schema.create(Schema.Type.INT);
+  }
+
+  public Schema getValueOutputSchema() {
+    return Schema.create(Schema.Type.STRING);
+  }
+
+  public TransformedRecord<Integer, String> put(Lazy<Integer> key, Lazy<String> value) {
```

Comment:
As of right now, the key in `TransformedRecord<K, V>` isn't in use. I wasn't part of the design discussion for this, so I'm not aware if a future use case has been identified.

Maybe @elijahgrimaldi can shine some light here?

Reply:
I believe I implemented this while we were having discussions about how we might want to do 1:N transformations but it was left incomplete and now it just wraps the serialization logic for the transformed value. It is excessive for what we have now so I think we could get rid of it.

Reply:
Yeah the original intention was to allow for this extension because.... well... at the time it seemed so simple :)

Reply:
I can go ahead and remove the key from `TransformedRecord`.

In [StoreIngestionTask](https://github.com/kvargha/venice/blob/88d834d26982e55c63da329bd708b8eacd812b1e/clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java#L3144-L3145), after `put` has been called, it calls `getValueBytes` on the `TransformedRecord` object. I'm thinking of moving the object creation logic out of the `put` function and right before `getValueBytes` is called. Does that sound good @FelixGV?

Reply:
I think we probably don't need the `TransformedRecord` class at all right now actually, if I remember correctly the original intent had to do with implementing some sort of output collector and all we really need is the serializer if we are transforming values 1:1

Reply:
I agree. We can do away with `TransformedRecord ` and move `getValueBytes` to `DaVinciRecordTransformer` as a default function.

Since we're no longer collecting output, I'm thinking we should rename `put` to `transformValue`, and maybe do away with the `delete` method. Thoughts?